### PR TITLE
fix todos, restructure services and fix some other mistakes

### PR DESCRIPTION
**Pull request by pablogh-2000**

### CODE REVIEW

File: tutorials/chapter1.md [hunk 1]
```diff
@@ -416,18 +414,32 @@ As it can be seen in this code, the ```kpsr::high_perf::DataMultiplexer``` has a
...
+There is an extensive use of the DataMultiplexer in the [vision tutorial](https://github.com/klepsydra-technologies/kpsr-vision-ocv-tutorial).
 
 ## Example 6: Working with services
 
 We are going to introduce now the API for ```kpsr::Service```, ```kpsr::Environment``` and ```kpsr::ManagedService```. These classes are not really necessary to benefit from the high performance API of Klepsydra, but might be of help.
 
 ### Service
 
-The ```kpsr::Service``` allows one to define a task which will run in the background, which can be administered remotely and/or which need to be run at regular intervals. The api expects us to define the custom start, stop, execute function as per the intended application. The public API functions to these are the startup(), shutdown() and runOnce().
+The ```kpsr::Service``` allows one to define a task which will run in the background, which can be administered remotely and/or which need to be run at regular intervals. The ```kpsr::Service``` class is used by means of inheritance. I.e., custome services must extend from this class. ```kpsr::Service``` has three pure virtual methods:
```
Comment (mharshe) [inline tutorials/chapter1.md]: custome --> custom

File: tutorials/chapter1.md [hunk 2]
```diff
@@ -416,18 +414,32 @@ As it can be seen in this code, the ```kpsr::high_perf::DataMultiplexer``` has a
...
+* ```stop()```, 
+* ```execute()``` 
+
+That custom services need to implement. These functions are never invoked directly, but via their public counterparts or public API functions:
+
+* ```startup()```, 
+* ```shutdown()``` and 
+* ```runOnce()```.
+
+Very important to know is that the ```runOnce``` invokes the ```execute``` method when the service is started, i.e., the ```startup``` function have ben called.
```
Comment (mharshe) [inline tutorials/chapter1.md]: i.e once the ```startup``` function has been called.

Comment (pablogh-2000) [reply tutorials/chapter1.md]: Fixed

File: tutorials/chapter1.md (hunk 1 above)
Comment (pablogh-2000) [reply tutorials/chapter1.md]: Fixed

**Pull request merged**